src/jit/lowerxarch.cpp

```diff
@@ -925,7 +925,8 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
     // the below logic leads to conservative register specification.  In future when or if we find
     // one such case, this logic needs to be fine tuned for that case(s).
     if (varTypeIsByte(tree) || ((tree->OperGet() == GT_CAST) && varTypeIsByte(tree->CastToType())) ||
-        (tree->OperIsCompare() && varTypeIsByte(tree->gtGetOp1()) && varTypeIsByte(tree->gtGetOp2())))
+        (tree->OperIsCompare() && varTypeIsByte(tree->gtGetOp1()) &&
+         (varTypeIsByte(tree->gtGetOp2()) || tree->gtGetOp2()->IsCnsIntOrI())))
```

Comment:
>       (varTypeIsByte(t [](start = 2, length = 23)

Please update the comment block above with an example IR because of which compare condition needs to be changed.